napari/utils/colormaps/colormap.py

```diff
@@ -230,3 +235,78 @@ def default_color(self):
         return self.color_dict.get(None, (0, 0, 0, 0))
         # we provided here default color for backward compatibility
         # if someone is using DirectLabelColormap directly, not through Label layer
+
+
+def cast_labels_to_minimum_type_auto(
+    data: np.ndarray, num_colors: int
+) -> np.ndarray:
+    """Perform modulo operation based on number of colors
+
+    Parameters
+    ----------
+    data : np.ndarray
+        Labels data to be casted.
+    num_colors : int
+        Number of unique colors in the data.
+
+    Returns
+    -------
+    np.ndarray
+        Casted labels data.
+    """
+    dtype = minimum_dtype_for_labels(num_colors + 1)
+
+    return _modulo_plus_one(data, num_colors, dtype)
+
+
+@numba.njit(parallel=True)
+def _modulo_plus_one(values: np.ndarray, n: int, dtype) -> np.ndarray:
+    """Like ``array % n`` but with 1 added to result for values >n.
+
+    This ensures (1) an output value in [0, n] (inclusive), and (2) that
+    no nonzero values in the input are zero in the output.
+
+    Parameters
+    ----------
+    values : np.ndarray
+        The dividend of the modulo operator.
+    n : int
+        The divisor.
+    dtype : np.dtype
+        The desired dtype for the output array.
+
+    Returns
+    -------
+    np.ndarray
+        The result: ``values`` for values in [0, n), ``values % n + 1``
+        everywhere else.
+    """
+    result = np.empty_like(values, dtype=dtype)
+
+    for i in numba.prange(values.size):
+        if 0 <= values.flat[i] < n:
```

Comment:
```suggestion
        if 0 <= values.flat[i] <= n:
```

for consistency with docstring

Reply:
@Czaki Ah, well the docstring is inconsistent with itself: the one-liner says `for values >n`, but the result description says `values in [0, n)` are not changed. So we can:
- update the oneliner to say `>=n`.
- Make this change and also update the result description.

What do we prefer? In the first scenario (and current code), `n-1` gets mapped to itself, `n` gets mapped to 1, `2*n - 1` gets mapped to `n`, and `2*n` gets mapped to 1. In the alternate scenario, `n` gets mapped to itself, and `n+1` gets mapped to 2. Either way, there is a discontinuity at one side of `n`.

Since `<` is slightly faster to evaluate than `<=` (? maybe this is no longer true with modern compilers), I suggest I just change the one-liner.

Reply:
I pushed up the updated description. But let me know if you want me to change it in the other direction.